client/pkg/transport/listener.go

```diff
@@ -423,10 +435,34 @@ func (info TLSInfo) baseConfig() (*tls.Config, error) {
 		}
 	}
 	if info.AllowedHostname != "" {
+		info.Logger.Warn("AllowedHostname is deprecated, use AllowedHostnames instead")
 		verifyCertificate = func(cert *x509.Certificate) bool {
 			return cert.VerifyHostname(info.AllowedHostname) == nil
 		}
 	}
+	if len(info.AllowedCNs) > 0 {
```

Comment:
I think those pairs should be also mutually exclusive:
* `AllowedHostnames` and `AllowedHostname`
* `AllowedCNs` and `AllowedCN`.

I the current code the new version will take precedence and override the `verifyCertificate` and `verifyCertificate` function.

Reply:
Yes, in the current code, AllowedCNs and AllowedHostnames will override AllowedCN and AllowedHostname.

What are you suggesting? Do we choose to report an error back when both configurations appear at the same time?

Reply:
Yes.